### PR TITLE
Adds java tcf proxy worker

### PR DESCRIPTION
**Pull request by asimonet**
The java worker listens to new work order events on a blockchain,
filters them to only keep those related to its local wallet, downloads
the appropriate docker container, executes it locally and returns the
result to a smart contract.

All TCF effort (establishing the secure channel, signing the result
etc.) must be done in the container.

Signed-off-by: Anthony Simonet-Boulogne <anthony@iex.ec>

### CODE REVIEW

**Comment by danintel** [comment]
This seems to be a duplicate/resubmital of PR #37

File: examples/common/java/src/main/java/org/eea/tcf/worker/chain/services/WorkOrderRegistryService.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        TransactionReceipt completeReceipt;
+        RemoteCall<TransactionReceipt> completeCall = getWorkOrderRegistryContract().workOrderComplete(
+                        workOrderIdBytes, BigInteger.valueOf(returnCode), response, signatureByte);
+
+        try {
+            completeReceipt = completeCall.send();
+            log.error("Work order complete [workOrderId:{}, returnCode:{}, gasUsed:{}]",
+                    workOrderId, returnCode, completeReceipt.getGasUsed());
+        } catch (Exception e) {
+            log.error("Failed to call WorkOrderComplete [workOrderId:{}, returnCode:{}, exeption:{}]",
```
Comment (danintel) [inline examples/common/java/src/main/java/org/eea/tcf/worker/chain/services/WorkOrderRegistryService.java]: From PR #37
s/exeption/exception/

File: examples/common/java/gradlew [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        (0) set -- ;;
+        (1) set -- "$args0" ;;
+        (2) set -- "$args0" "$args1" ;;
+        (3) set -- "$args0" "$args1" "$args2" ;;
+        (4) set -- "$args0" "$args1" "$args2" "$args3" ;;
+        (5) set -- "$args0" "$args1" "$args2" "$args3" "$args4" ;;
+        (6) set -- "$args0" "$args1" "$args2" "$args3" "$args4" "$args5" ;;
+        (7) set -- "$args0" "$args1" "$args2" "$args3" "$args4" "$args5" "$args6" ;;
+        (8) set -- "$args0" "$args1" "$args2" "$args3" "$args4" "$args5" "$args6" "$args7" ;;
+        (9) set -- "$args0" "$args1" "$args2" "$args3" "$args4" "$args5" "$args6" "$args7" "$args8" ;;
```
Comment (danintel) [inline examples/common/java/gradlew]: From PR #37 

> Ram-srini 27 days ago • 
> What do these args indicate?
> Could you please add description about these args in the comment.

Comment (asimonet) [reply examples/common/java/gradlew]: This code is generated by the Gradle build system to allow the project to be built out of the box. Changing it is not recommended.

File: examples/common/java/src/main/java/org/eea/tcf/worker/chain/services/WorkOrderRegistryService.java (hunk 1 above)
Comment (asimonet) [reply examples/common/java/src/main/java/org/eea/tcf/worker/chain/services/WorkOrderRegistryService.java]: This one slipped my mind, I fixed it in a new commit.

**Comment by asimonet** [comment]
About the license: I added the standard format to all Java files, should I add it somewhere else?

File: examples/common/java/direct/worker/src/main/resources/wallet/wallet1.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "address": "1a69b2eb604db8eba185df03ea4f5288dcbbd248",
+  "id": "9428da60-eafb-4771-a64b-fac009946bba",
+  "version": 3,
+  "Crypto": {
+    "cipher": "aes-128-ctr",
```
Comment (manojgop) [inline examples/common/java/direct/worker/src/main/resources/wallet/wallet1.json]: You might want to change it to aes-256-gcm which is supported by TCF

Comment (asimonet) [reply examples/common/java/direct/worker/src/main/resources/wallet/wallet1.json]: This wallet is just for testing, we can update to test different ciphers in the future.

File: examples/common/java/direct/worker/src/main/java/org/eea/tcf/worker/chain/services/Web3jService.java [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    private EthBlock.Block getBlock(long blockNumber) throws IOException {
+        return web3j.ethGetBlockByNumber(DefaultBlockParameter.valueOf(BigInteger.valueOf(blockNumber)),
+                false).send().getBlock();
+    }
+
+    // check if the blockNumber is already available for the scheduler
+    // blockNumber is different than 0 only for status the require a check on the blockchain, so the scheduler should
+    // already have this block, otherwise it should wait for a maximum of 10 blocks.
+    public boolean isBlockAvailable(long blockNumber) {
+        // if the blocknumer is already available then simply returns true
```
Comment (danintel) [inline examples/common/java/direct/worker/src/main/java/org/eea/tcf/worker/chain/services/Web3jService.java]: s/blocknumer/blocknumber/

**Pull request merged**